### DISCUSSION TITLE
Solve one field for single time step in multi-field problem

### DISCUSSION THREAD

**Discussion by avtarsinghh1991** [idaholab/moose#19994] [Q&A Modules: General] [answered]
Hello MOOSE Experts

I am solving the coupled multi-field problem. I want to solve any one or two fields only for the first time step and the rest of the variables for all the given time steps. Can you please let me know what is the procedure to do it with a suitable example, if possible?
I shall be very thankful.

For example:

Let's say I have four fields
c (diffusion, time-dependent), u (displacement), eta_1 and eta_2

I am solving Helmholtz equation to find eta_1 and eta_2.

I want to solve eta_1 and eta_2 for the first step only and use the obtained values for the rest of the simulation.

If it's possible, eta_1 and eta_2 can also be solved even prior to the beginning of the simulation.
Then use those values for the rest of the simulation.

Please help.

Best
Avtar

**Comment by GiudGiud** [comment, marked as answer]
Hello,

You will want to use a Multiapp setup for this, then transfer the two fields that were solved for in the beginning to the other solve. 
One subapp will solve for eta_1/2 and only run in the beginning (execute_on = INITIAL) and the other, the main app, will solve for the other variables, after having retrieved eta_1/2 from the subapp using a transfer.

More info on Multiapps:
https://mooseframework.inl.gov/getting_started/examples_and_tutorials/tutorial02_multiapps/index.html

SInce you are solving for eta_1/2 only once you will want to use a FullSolveMultiApp
https://mooseframework.inl.gov/source/multiapps/FullSolveMultiApp.html

Guillaume
